pages/component.md

```diff
@@ -8,14 +8,402 @@ hideInToc: true
+
+#### In React, components often have a <span class="text-teal-400">parent-child relationship</span>, where the **parent** component contains other components as its **children**. This relationship allows for **data** to flow between components, enabling a <span class="text-teal-400">structured</span>, <span class="text-teal-400">reusable</span>, and <span class="text-teal-400">maintainable</span> UI.
+
+</div>
+
+<br/>
+
+## **Think of that House 🏠 again**
+
+A room (parent) may contain **furniture** (children) like a 🛏️ bed, 🪑 chair, or 💡 lamp. Each furniture item (child) serves its specific role, but the room (parent) coordinates their placement and usage.
+
+<p v-click>Similarly, in React, a <u class="text-teal-400">parent component collaborates with its children to build a cohesive application</u>. The previously discussed nested components 🔙 are the children of the App (parent) component.</p>
+
+<p v-click> - A parent component renders one or more child components within it. It can pass data and functions to its children. </p>
+
+<p v-click> - A child component is nested within a parent component. It can receive data from the parent to render its UI or perform actions. </p>
+
+---
+hideInToc: true
+transition: slide-down
+---
+
+<h1 class="text-gradient text-4xl">Props <span class="text-common">📨</span>: Passing and Reading Data Between Components</h1>
+
+#### Props (short for properties) allow parent components to pass data to their child components, just like <u>arguments in functions</u> and <u>attributes in html</u>. <span class="text-teal-400">You can pass any JavaScript value as a prop</span> (objects, arrays, functions, etc.). Below, child compoent `Welcome` is receiving name props from parent component `App`
+
+<v-click>
+
+```jsx
+function App() {
+  return (
+    <div>
+      <Welcome name="Alice" />
+      <Welcome name="Bob" />
+    </div>
+  )
+}
+
+function Welcome(props) {
+  return <h1>Hello, {props.name}!</h1>
+}
+
+// Output:
+// Hello, Alice!
+// Hello, Bob!
+```
+
+</v-click>
+
+---
+hideInToc: true
+transition: slide-left
+---
+
+# How to Use `Props`
+
+<div class="flex gap-4">
+<div v-click >
+ 1️⃣ Pass Props to Components: Add them like HTML attributes.
+
+```jsx
+function Profile() {
+  return (
+    <div>
+      <Avatar person={{ name: 'Ojo Setemi', imageId: '1bX5QH6' }} />
+    </div>
+  )
+}
+```
+
+</div>
+
+<div v-click> 
+2️⃣ Destructure Props: Extract props directly in the function parameters, with the option to assign default values.
+
+```jsx
+function Avatar({ person, size = 100 }) {
+  return (
+    <img
+      src={`https://i.imgur.com/${person.imageId}.jpg`}
+      alt={person.name}
+      width={size}
+      height={size}
+    />
+  )
+}
+```
+
+</div>
+</div>
+
+<div v-click>
+Forwarding Props with Spread Syntax: You can forward all props to child components with ...props
+
+```jsx
+function Profile() {
+  return (
+    <div>
+      <Avatar {...props} />
+    </div>
+  )
+}
+```
+
+</div>
```

Comment:
The content on the right hand side of the slide are partly cut off on my screen, same for the bottom div. Consider restructuring the entire slide or moving some items to another slide. Although reducing the div sizes can also fix the overflows while maintaining the same style.

Reply:
Thank you. I adjusted the codes already. Please check. Thank you.